assets/src/components/activities/adaptive/components/authoring/LayoutEditor.tsx

```diff
@@ -139,7 +134,7 @@ const LayoutEditor: React.FC<LayoutEditorProps> = (props) => {
         props.hostRef.removeEventListener('click', handleHostClick);
       }
     };
-  }, [props, parts, configurePartId]);
+  }, [props.id, props.hostRef, parts, configurePartId]);
```

Comment:
I could be wrong, but I think adding `props` as a dependency would trigger this effect every single render cycle.